File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int idx = 0;
        int temp;

        for(int i = 0; i<s.length(); i++){
            temp = t.indexOf(s.charAt(i), idx);
            System.out.println(temp);
            if(temp>-1){
                idx= Math.max(temp+1, idx);
            }
            else{
                return false;
            }
        }

        return true;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) > len(t):
            return False

        idx1 = 0
        idx2 = 0
        print(s, t)
        while idx1<len(s) and idx2<len(t):
            
            temp1 = s[idx1] if idx1<len(s) else ""
            temp2 = t[idx2] if idx2<len(t) else ""

            if temp1==temp2:
                idx1+=1
                idx2+=1
            else:
                idx2+=1
            
        return idx1 == len(s)